streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -687,7 +687,8 @@ public static StreamThread create(final InternalTopologyBuilder builder,
             builder,
             threadClientId,
             logContext,
-            assignmentErrorCode);
+            assignmentErrorCode)
+            .updateThreadMetadata(getSharedAdminClientId(clientId));
```

Comment:
Why do we not pass the admin client-id into the constructor?

Reply:
I've thought about passing in the client-id, it is a bit overkill in addition to the thread-client-id just to set the shared admin client inside the constructor. I felt the current way is less intrusive but if you feel strong the other way we can discuss.

Reply:
Because the shared admit client id does never change, it seems weird to have a setter that allow to update is at any time. It seems cleaner to pass it in the constructor and make it immutable this way.